src/iuse_actor.cpp

```diff
@@ -2578,6 +2609,17 @@ repair_item_actor::attempt_hint repair_item_actor::repair( player &pl, item &too
         roll = NEUTRAL;
     }
 
+    if (t->has_quality(quality_sew)) {
+        p->add_msg_if_player(m_info,
+            _("You have used %d thread charges."), roll == SUCCESS ? filament_required : 1);
+        for (const auto &it : roll == SUCCESS ? req_success.get_components() : req.get_components() ) {
+            p->consume_items(it);
```

Comment:
Try this with more than one thread type. For example, one thread in inventory and one on the floor and also yarn on the floor.
If this asks for thread type more than once, it is unacceptable.